File: task/task_upload.ts
import { TwelveLabs, Task } from 'twelvelabs-js';

(async () => {
  const client = new TwelveLabs({ apiKey: process.env.API_KEY}); 
  const task = await client.task.create({
    indexId: '',
    file: '',
    // transcriptionFile: '<YOUR_TRANSCRIPTION_FILE>'),
  });
  console.log(`Task ID:${task.id}`);
  await task.waitForDone(500, (task: Task) => {
    console.log(`  Status=${task.status}`);
  });
  
  if (task.status !== 'ready') {
    throw new Error(`Indexing failed with status ${task.status}`);
  }
  console.log(`Video ID: ${task.videoId}`);


})();

File: task/task_upload.py
from twelvelabs import TwelveLabs
from twelvelabs.models.task import Task
import os

client = TwelveLabs(api_key=os.environ.get("API_KEY"))

task = client.task.create(
  index_id="",
  file="",
  # transcription_file="<YOUR_TRANSCRIPTION_FILE>"
)
print(f"Task id={task.id}")

# Utility function to print the status of a video indexing task
def on_task_update(task: Task):
      print(f"  Status={task.status}")

task.wait_for_done(sleep_interval=5, callback=on_task_update)

if task.status != "ready":
  raise RuntimeError(f"Indexing failed with status {task.status}")
print(f"Video ID: {task.video_id}")
